### PR TITLE
GH-501 adding refresh token functionality to handle edge case of 

### PR DESCRIPTION
**Pull request by fokusferit**
adding refresh token functionality to handle edge case of outdated access token.

I'm creating this PR in favor of #556 as that one is highly outdated and merging all stuff would need time. And we are creating a docker image which can be tested in our testbed from @rashamalek 

Still missing:

- Proper Tests (FE and BE)
- Visualize Refresh Token Button State (indicate that things are in Progress)
- Align Button more properly

### CODE REVIEW

**Comment by rashamalek** [comment]
:+1:

File: client/actions/checks.js [hunk 1]
```diff
@@ -73,9 +79,9 @@ function updateTokenForChecks(repo){
  * Refresh access token with new user logged in (1% case)
  * @param {*} repo - repository data
  */
-export function requestRefreshToken(repo) {
+export function requestRefreshToken(repo, reloadReposFn) {
```
Comment (fokusferit) [inline client/actions/checks.js]: Might be interesting to ask someone like @frometor or @ChristianLohmann regarding Redux:
- In the Component I map the action to props and then provide it to my action here. 
-> I don't see really another easy way of triggered a reload of all repo's and the data as the function `reloadIfNeeded` is not bound to any state parameter just function arguments...

**Comment by rashamalek** [comment]
@fokusferit the previous feature "prevent toggling the specification checks" is not working on zappr-test, could you verify?
by clicking on specification check I expected to receive that red alert saying github branch protection setting is enabled, and prevent from the button toggle, but it just toggles and then toggling back shows :
screenshot 2019-03-06 20 22 05

**Comment by scherniavsky** [comment]
@fokusferit @ChristianLohmann @harti2006 how important is to merge this PR before turning the Branch protection on?

**Comment by fokusferit** [comment]
@scherniavsky I can't say directly how blocking it is what I can say (based on my information) is that the moment you turn on branch protection is when teams can't update the access tokens of zappr checks because of team member leaving the company and the token expires or generally when access token expire.

Just to understand the context:
- There is no service to service token which can be used globally within Zappr. For each check the access token of the user who enabled it is stored, therefore when a user is not valid (leaves the company) the check is not able to run.
The implication is that some teams will be blocked, create a Support tickets for Automata / Machinery I guess and they need to do something manually (but idk what and how this works.)

My knowledge is a bit limited. I don't know how much of support requests we are talking about. The branch protection can be enabled and the zappr supports that (if the current master is already deployed and live).

**Comment by lotharschulz** [comment]
https://developer.github.com/apps/differences-between-apps/#token-based-identification may help to add some context with respect to 

> There is no service to service token which can be used globally within Zappr. For each check the access token of the user who enabled it is stored, therefore when a user is not valid (leaves the company) the check is not able to run.

**Comment by rashamalek** [comment]
:+1:

**Comment by fokusferit** [comment]
👍

**Pull request merged**